reference/enterprise-api-ref/ldap-api.markdown

```diff
@@ -0,0 +1,125 @@
+---
+layout: default
+title: LDAP authentication API
+published: true
+tags: [reference, enterprise, API, authenticating, LDAP]
+---
+
+LDAP authentication API allows to check ldap user credentials and change LDAP settings.
+
+## Login
+
+**URI:** https://hub.cfengine.com/ldap/login
+
+**Method:** POST
+
+**Parameters:**
+
+* **username** *(string)*
+    Username from LDAP
+* **password** *(string)*
+    User password
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "success": true,
+    "message": "You are successfully authenticated"
+}
+```
+
+## Get settings
+
+**URI:** https://hub.cfengine.com/ldap/settings
+
+**Method:** GET
+
+**Headers:**
+
+* **Authorization: api_token** *(string)*
+    Set token to access api
+
+**Example response:**
+
+```
+HTTP 200 Ok
+{
+    "success": true,
+    "data": {
+        "domain_controllers": "local.loc",
+        "base_dn": "DC=local,DC=loc",
+        "login_attribute": "cn",
+        "port": 365,
+        "use_ssl": false,
+        "use_tls": false,
+        "timeout": 5,
+        "admin_username": "cn=admin,DC=local,DC=loc",
+        "admin_password": "Password is set"
+    }
+}
+```
+
+**Output:**
+
+* **domain_controllers**
+    The domain controllers option is   server name located on your network that serve Active Directory.
+* **base_dn**
+    The base distinguished name is the base distinguished name you'd like to perform operations on. An example base DN would be DC=corp,DC=acme,DC=org.
+* **login_attribute**
+    Login attribute like cn or uid
+* **port**
+    The port option is used for authenticating and binding to your AD server. The default ports are already used for non SSL and SSL connections (389 and 636).
+* **use_ssl**
+    User ssl for connection
+* **use_tls**
+    User tls for connection
+* **timeout**
+    The timeout option allows you to configure the amount of seconds to wait until your application receives a response from your LDAP server.
+* **admin_username**
+    LDAP admin distinguished name
+* **admin_password**
+    LDAP admin password.
+
+## Update settings
+
+**URI:** https://hub.cfengine.com/ldap/settings
+
+**Method:** PATCH
+
+**Headers:**
+
+* **Authorization: api_token** *(string)*
+    Set token to access api
```

Comment:
How does one get this API token?

Reply:
API token i /ldap/config/settings.php https://github.com/cfengine/ldap-api/blob/master/config/settings.php#L17 .

Reply:
I am not sure about posting this documentation as public... Because it's more internal stuff. And using as separate part makes no sense. I this we will discuss it.

Reply:
I've added info about token.